bin/inx-chronicle/src/api/v2/routes.rs

```diff
@@ -0,0 +1,641 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use axum::{
+    extract::{Extension, Path},
+    routing::*,
+    Router,
+};
+use chronicle::{
+    bson::{BsonExt, DocExt, U64},
+    db::model::inclusion_state::LedgerInclusionState,
+    stardust::{
+        output::OutputId,
+        payload::transaction::{TransactionId, TransactionPayload},
+    },
+};
+use futures::TryStreamExt;
+use mongodb::{
+    bson::{doc, DateTime, Document},
+    options::FindOptions,
+    Database,
+};
+use time::OffsetDateTime;
+
+use super::{
+    extractors::{MessagesQuery, OutputsQuery},
+    responses::*,
+};
+use crate::api::{
+    error::APIError,
+    extractors::{Expanded, Pagination, TimeRange},
+    APIResult,
+};
+
+pub fn routes() -> Router {
+    Router::new()
+        .nest(
+            "/messages",
+            Router::new()
+                .route("/", get(messages_query))
+                .route("/:message_id", get(message))
+                .route("/:message_id/raw", get(message_raw))
+                .route("/:message_id/metadata", get(message_metadata))
+                .route("/:message_id/children", get(message_children)),
+        )
+        .nest(
+            "/outputs",
+            Router::new()
+                .route("/", get(outputs_query))
+                .route("/:transaction_id/:idx", get(output)),
+        )
+        .nest(
+            "/transactions",
+            Router::new()
+                .route("/:message_id", get(transaction_for_message))
+                .route("/history/:address", get(transaction_history))
+                .route("/included-message/:transaction_id", get(transaction_included_message)),
+        )
+        .route("/milestones/:index", get(milestone))
+        .nest("/analytics", Router::new().route("/addresses", get(address_analytics)))
+}
+
+async fn message(database: Extension<Database>, Path(message_id): Path<String>) -> APIResult<MessageResponse> {
+    let mut rec = database
+        .collection::<Document>("stardust_messages")
+        .find_one(doc! {"message_id": &message_id}, None)
+        .await?
+        .ok_or(APIError::NoResults)?;
+    let mut message = rec.take_document("message")?;
+    Ok(MessageResponse {
+        protocol_version: message.get_as_u8("protocol_version")?,
+        parents: message
+            .take_array("parents")?
+            .iter()
+            .map(|m| m.as_string())
+            .collect::<Result<_, _>>()?,
+        payload: message.take_bson("payload").ok().map(Into::into),
+        nonce: message.convert_document::<U64, _>("nonce")?.into(),
+    })
```

Comment:
Should we add a level of indirection between the database and the API, like we do with inserting messages (could also be done in a later PR)?